.github/workflows/release.yml

```diff
@@ -19,18 +16,69 @@ jobs:
     - name: Setup Go
       uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491
       with:
-        go-version: '1.22.1' 
+        go-version: '1.22.2' 
     
     - name: Build and Run
       run: |
         go build -o festerize main.go
-        ./festerize 
-    - name: Upload Release Assets
-      uses: actions/upload-release-asset@e8f9f06c4b078e705bd2ea027f0926603fc9b4d5
-      env: 
+
+    # Zip binary for Ubuntu
+    - name: Zip binary
+      run: zip festerize_ubuntu.zip festerize
+
+    - name: Upload Ubunutu Release Assets
+      uses: softprops/action-gh-release@9d7c94cfd0a1f3ed45544c887983e9fa900f0564
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        files: festerize_ubuntu.zip
+
+  build_and_upload_mac:
+    runs-on: macos-latest
+
+    steps:
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+    
+    # Setup Go
+    - name: Setup Go
+      uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491
+      with:
+        go-version: '1.22.2' 
+    
+    - name: Build and Run
+      run: |
+        go build -o festerize main.go
+    # Zip binary for Mac
+    - name: Zip binary
+      run: zip festerize_mac.zip festerize 
+    - name: Upload Mac Release Assets
+      uses: softprops/action-gh-release@9d7c94cfd0a1f3ed45544c887983e9fa900f0564
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        files: festerize_mac.zip
+
+  build_and_upload_windows:
+    runs-on: windows-latest
+
+    steps:
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+    
+    # Setup Go
+    - name: Setup Go
+      uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491
```

Comment:
It might be nice to add a comment with the version represented by the hash, so we know what our version status is with a quick glance. An example: https://github.com/UCLALibrary/service-template/blob/dcb5afece976642251228347ad16c3a859bcb008/.github/workflows/build.yml#L28